### ISSUE TITLE
test_localhost(TestResolver) fails

### ISSUE THREAD

**Issue by voxik** [eventmachine/eventmachine#579]
I am observing followint test in EM 1.0.6, running by Ruby 2.2:

```
Failure: <false> is not true.
test_localhost(TestResolver)
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/tests/test_resolver.rb:46:in `block (2 levels) in test_localhost'
     43:   def test_localhost
     44:     EM.run {
     45:       d = EM::DNS::Resolver.resolve "localhost"
  => 46:       d.errback { assert false }
     47:       d.callback { |r|
     48:         assert_include(["127.0.0.1", "::1"], r.first)
     49:         assert_kind_of(Array, r)
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/em/deferrable.rb:158:in `call'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/em/deferrable.rb:158:in `set_deferred_status'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/em/deferrable.rb:198:in `fail'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/em/resolver.rb:174:in `receive_answer'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/em/resolver.rb:128:in `receive_data'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/eventmachine.rb:187:in `run_machine'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/lib/eventmachine.rb:187:in `run'
/builddir/build/BUILD/rubygem-eventmachine-1.0.6/usr/share/gems/gems/eventmachine-1.0.6/tests/test_resolver.rb:44:in `test_localhost'
```

**Comment by sodabrew** [comment]
Oh, I've never seen this issue before. If you run `ping localhost` on the command line what do you get?

**Comment by sodabrew** [comment]
I wonder if this is the same issue as #439 where, for whatever reason, you don't have a `localhost` alias in /etc/hosts?

**Comment by voxik** [comment]
ping works just fine:

```
# ping localhost
PING localhost.localdomain (127.0.0.1) 56(84) bytes of data.
64 bytes from localhost.localdomain (127.0.0.1): icmp_seq=1 ttl=64 time=0.031 ms
64 bytes from localhost.localdomain (127.0.0.1): icmp_seq=2 ttl=64 time=0.066 ms
^C
--- localhost.localdomain ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 999ms
rtt min/avg/max/mdev = 0.031/0.048/0.066/0.018 ms
```

and this is content of my /etc/hosts:

```
# cat /etc/hosts
127.0.0.1		localhost.localdomain localhost
::1		localhost6.localdomain6 localhost6tes
```

1.0.6 already contains the #439, so this should be something different.

Please note that I am executing the test suite in mock, which is used by Fedora's builders and prepares just minimal chrooted environment to build package.

**Comment by sodabrew** [comment]
Do you see the same problem with EM 1.0.7? Nothing changed that should have affected this issue, but I'd be interested to double check. Also 1.0.7 has an important memory leak fix.

**Comment by voxik** [comment]
I still observe the issue with 1.0.8 (sorry for the slow response :/)

**Comment by voxik** [comment]
Still issue with 1.2.1.

**Comment by sodabrew** [comment]
> Please note that I am executing the test suite in mock, which is used by Fedora's builders and prepares just minimal chrooted environment to build package.

Does the chroot have an /etc/hosts or an /etc/resolv.conf?

**Comment by voxik** [comment]
I listed the /etc/hosts above. This is the resolv.conf:

~~~
$ cat /etc/resolv.conf 
# This is a static resolv.conf file for connecting local clients to
# systemd-resolved via its DNS stub listener on 127.0.0.53.
#
# Third party programs must not access this file directly, but only through the
# symlink at /etc/resolv.conf. To manage resolv.conf(5) in a different way,
# replace this symlink by a static file or a different symlink.
#
# See systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 127.0.0.53
~~~

**Comment by voxik** [comment]
Interestingly, I cannot reproduce the issue anymore, trying older versions as well as more recent versions ...
